src/dodrio_berry_picking_2.c

```diff
@@ -1,23 +1,1617 @@
+{
+    void *ptr = AllocZeroed(0x480);
+    struct SpritePalette sprPal = {sDodrioBerrySpritesPal, 3};
+
+    LZ77UnCompWram(sDodrioBerrySpritesGfx, ptr);
+    if (ptr != NULL)
+    {
+        struct SpriteSheet sprSheet = {ptr, 0x480, 2};
+        LoadSpriteSheet(&sprSheet);
+    }
+
+    LoadSpritePalette(&sprPal);
+    Free(ptr);
+}
+
+static const s16 sUnknown_8478DD4[] = {88, 128, 168, 208};
+
+void sub_815417C(void)
+{
+    u8 i;
+    s16 x;
+
+    struct SpriteTemplate sprTemplate1 =
+    {
+        .tileTag = 2,
+        .paletteTag = 3,
+        .oam = &sOamData_8478CA8,
+        .anims = sSpriteAnimTable_8478D60,
+        .images = NULL,
+        .affineAnims = gDummySpriteAffineAnimTable,
+        .callback = SpriteCallbackDummy,
+    };
+    struct SpriteTemplate sprTemplate2 =
+    {
+        .tileTag = 2,
+        .paletteTag = 3,
+        .oam = &sOamData_8478CA0,
+        .anims = sSpriteAnimTable_8478D60,
+        .images = NULL,
+        .affineAnims = gDummySpriteAffineAnimTable,
+        .callback = SpriteCallbackDummy,
+    };
+
+    for (i = 0; i < 11; i++)
+    {
+        gUnknown_203F400[i] = AllocZeroed(4);
+        x = i * 16;
+        *gUnknown_203F400[i] = CreateSprite(&sprTemplate1, x + (i * 8), 8, 1);
+        sub_81542EC(i, TRUE);
+    }
+    for (i = 0; i < 4; i++)
+    {
+        gUnknown_203F42C[i] = AllocZeroed(4);
+        if (i == 3)
+            *gUnknown_203F42C[i] = CreateSprite(&sprTemplate2, sUnknown_8478DD4[i], 57, 0);
+        else
+            *gUnknown_203F42C[i] = CreateSprite(&sprTemplate2, sUnknown_8478DD4[i], 60, 0);
+        StartSpriteAnim(&gSprites[*gUnknown_203F42C[i]], i);
+    }
+
+    sub_8154324(TRUE);
+}
+
+void sub_8154274(void)
+{
+    struct Sprite * sprite;
+    u8 i;
+
+    for (i = 0; i < 11; i++)
+    {
+        sprite = &gSprites[*gUnknown_203F400[i]];
+        if (sprite != NULL)
+            DestroySprite(sprite);
+        FREE_AND_SET_NULL(gUnknown_203F400[i]);
+    }
+    for (i = 0; i < 4; i++)
+    {
+        sprite = &gSprites[*gUnknown_203F42C[i]];
+        if (sprite != NULL)
+            DestroySprite(sprite);
+        FREE_AND_SET_NULL(gUnknown_203F42C[i]);
+    }
+}
+
+void sub_81542EC(u8 id, bool8 invisible)
+{
+    gSprites[*gUnknown_203F400[id]].invisible = invisible;
+}
+
+static void sub_8154324(bool8 invisible)
+{
+    u8 i;
+    for (i = 0; i < 4; i++)
+        gSprites[*gUnknown_203F42C[i]].invisible = invisible;
+}
+
+void sub_8154370(u8 id, u8 y)
+{
+    gSprites[*gUnknown_203F400[id]].pos1.y = y * 8;
+}
+
+void sub_8154398(u16 id, u8 frameNum)
+{
+    StartSpriteAnim(&gSprites[*gUnknown_203F400[id]], frameNum);
+}
+
+// Unused
+static void sub_81543C4(u8 spriteId)
+{
+    gSprites[spriteId].pos1.x = 20 * spriteId + 50;
+    gSprites[spriteId].pos1.y = 50;
+}
+
+// Gamefreak made a mistake there and goes out of bounds for the data array as it holds 8 elements
+// in turn overwriting sprite's subpriority and subsprites fields.
+#if defined(NONMATCHING)
```

Comment:
probably we can make another macro for known bug fixes, as I feel NONMATCHING is not serving for the purpose